packages/@lwc/engine-core/src/framework/vm.ts

```diff
@@ -99,6 +99,8 @@ export interface Context {
     wiredDisconnecting: Array<() => void>;
 }
 
+export type RefVNodes = { [name: string]: VNode };
```

Comment:
Only elements can be referenced by refs. Not nodes.

```suggestion
export type RefVNodes = { [name: string]: VBaseElement };
```

This would allow removing the [type guard](https://github.com/salesforce/lwc/pull/2690/files#diff-f6642700b605b151bc31ff4e3a99bdd950049dde971421a25d0f0c4e7fe90d12R495) in `base-lightning-element.ts`.